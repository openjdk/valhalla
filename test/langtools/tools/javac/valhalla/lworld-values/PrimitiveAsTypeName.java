/*
 * @test /nodynamiccopyright/
 * @bug 8267821
 * @summary Javac's handling of `primitive' modifier is unlike the handling of other restricted identifiers
 * @compile/fail/ref=PrimitiveAsTypeName.out --source 16 -XDrawDiagnostics PrimitiveAsTypeName.java
 */

public class PrimitiveAsTypeName {
    public class primitive {
        primitive x;
        primitive foo(int l) {}
    }
}
