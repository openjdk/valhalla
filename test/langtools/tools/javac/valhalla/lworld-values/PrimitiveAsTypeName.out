- compiler.warn.source.no.system.modules.path: 16
PrimitiveAsTypeName.java:9:18: compiler.warn.restricted.type.not.allowed.preview: primitive, 17
PrimitiveAsTypeName.java:10:9: compiler.warn.restricted.type.not.allowed.preview: primitive, 17
PrimitiveAsTypeName.java:11:9: compiler.warn.restricted.type.not.allowed.preview: primitive, 17
PrimitiveAsTypeName.java:11:31: compiler.err.missing.ret.stmt
1 error
4 warnings
